modules/tts/tts-adapters/tts-adapter-sapinative/src/test/java/org/daisy/pipeline/tts/sapi/impl/OnecoreTest.java

```diff
@@ -0,0 +1,240 @@
+package org.daisy.pipeline.tts.sapi.impl;
+
+import java.util.Collection;
+import java.util.HashMap;
+
+import org.daisy.pipeline.tts.onecore.Onecore;
+import org.daisy.pipeline.tts.TTSService.SynthesisException;
+import org.daisy.pipeline.tts.Voice;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class OnecoreTest {
+
+	@BeforeClass
+	public static void load() throws SynthesisException {
+		SAPIService.loadOnecoreDLL();
+		Onecore.initialize();
+	}
+
+	@AfterClass
+	public static void dispose() {
+		System.out.println("WARNING notice: you may encountered an exception when the dll is released by the JVM after finishing tests, with an error log generated in the project folder.");
+		System.out.println("This exception does not prevent the tests and build to complete in success, and has not yet occured in real-world production tests.");
+		System.out.println("You can ignore it and the error log.");
+		Onecore.dispose();
+	}
+
+	@Test
+	public void getVoiceNames() {
+		String[] voices = Onecore.getVoiceNames();
+		Assert.assertTrue(voices.length > 0);
+	}
+
+	@Test
+	public void getVoiceVendors() {
+		String[] vendors = Onecore.getVoiceVendors();
+		Assert.assertTrue(vendors.length > 0);
+	}
+
+	@Test
+	public void getVoiceLocales() {
+		String[] locales = Onecore.getVoiceLocales();
+		Assert.assertTrue(locales.length > 0);
+	}
+
+	@Test
+	public void getVoiceGenders() {
+		String[] genders = Onecore.getVoiceGenders();
+		Assert.assertTrue(genders.length > 0);
+	}
+
+	@Test
+	public void getVoiceAges() {
+		String[] ages = Onecore.getVoiceAges();
+		Assert.assertTrue(ages.length > 0);
+	}
+
+	@Test
+	public void manageConnection() {
+		long connection = Onecore.openConnection();
+		Assert.assertNotSame(0, connection);
+		Onecore.closeConnection(connection);
+	}
+
+	@Test
+	public void getVoiceInfo() throws Throwable {
+		Collection<Voice> voices = allocateEngine().getAvailableVoices();
+		Assert.assertTrue(voices.size() > 1);
+	}
+
+	// https://github.com/microsoft/Windows-universal-samples/tree/main/Samples/SpeechRecognitionAndSynthesis
+	// <speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis'
+	// xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'
+	// xsi:schemaLocation='http://www.w3.org/2001/10/synthesis  http://www.w3.org/TR/speech-synthesis/synthesis.xsd'
+	//"<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis'><s>" + x + "</s></speak>";
+	// <ssml:speak xmlns:ssml="http://www.w3.org/2001/10/synthesis" version="1.0"><s:s xmlns:tmp="http://" xmlns:s="http://www.w3.org/2001/10/synthesis" id="s1">small sentence</s:s><ssml:break time="250ms"/></ssml:speak>
```

Comment:
What is this?

Reply:
Oups forgot to clean that up.
I was keeping traces of the ssml used in MS documentation to check what was wrong with our ssml.
I'll remove that.

Reply:
OK cool.

Reply:
Fixed.